docs/deploy/agent-engine/deploy.md

```diff
@@ -0,0 +1,188 @@
+
+6. **Create a Google Cloud Storage (GCS) Bucket**:
+    * Agent Engine requires a GCS bucket to stage your agent's code and dependencies for deployment. If already have one, we recommend you create a new one.
+    * Create a GCS bucket by following the instructions [here](https://cloud.google.com/storage/docs/creating-buckets). We recommend starting with the default settings when creating your first bucket.
+    * Once you have created a storage bucket, you should be able to see it on the [Cloud Storage Buckets page](https://console.cloud.google.com/storage/browser).
+    * You need the GCS bucket path to set as your staging bucket. For example, if your GCS bucket name is "my-bucket", then your bucket path should be "gs://my-bucket".
+
+## Set up your coding environment {#prerequisites-coding-env}
+
+Now that you prepared your Google Cloud project, you can return to your coding
+environment. These steps require access to a terminal within your coding
+environment to run command line instuctions.
+
+### Authenticate your coding environment with Google Cloud
+
+*   You need to authenticate your coding environment so that you and your
+    code can interact with Google Cloud. To do so, you need the gcloud CLI.
+    If you have never used the gcloud CLI, you need to first
+    [download and install it](https://docs.cloud.google.com/sdk/docs/install-sdk)
+    before continuing with the steps below:
+
+*   Run the following command in your terminal to access your Google Cloud
+    project as a user:
+
+    ```shell
+    gcloud auth login
+    ```
+
+    You should now see "You are now authenticated with the gcloud CLI!".
+
+*   Run the following command to authenticate your code so that it can work with
+    Google Cloud:
+
+    ```shell
+    gcloud auth application-default login
+    ```
+
+    You should now see "You are now authenticated with the gcloud CLI!".
+
+*   (Optional) If you need to set or change your default project in gcloud, you
+    can use:
+
+    ```shell
+    gcloud config set project MY-PROJECT-ID
+    ```
+
+### Define your agent {#define-your-agent}
+
+With your Google Cloud and coding environment prepared, you're ready to deploy
+your agent. The instructions assume that you have an agent project folder,
+such as:
+
+```shell
+multi_tool_agent/
+├── .env
+├── __init__.py
+└── agent.py
+```
+
+You can also refer to the sample agent on Github: [multi_tool_agent](https://github.com/google/adk-docs/tree/main/examples/python/snippets/get-started/multi_tool_agent).
+
+## Deploy the agent {#deploy-agent}
+
+You can deploy from your terminal using the `adk deploy` command line tool. This
+process packages your code, builds it into a container, and deploys it to the
+managed Agent Engine service. This process can take several minutes.
+
+### adk deploy agent_engine
```

Comment:
This heading is odd to read as a deployment command. Should we rename it to something like "Run deploy command"?